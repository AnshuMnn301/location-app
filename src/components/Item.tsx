import React from 'react';
import {View, Text, Button} from 'react-native';
import {ThemeContext} from '../../App';
import {LocationType} from '../utils/apicalls';

type Props = {
  item: LocationType;
  deleteItem: Function;
  section: {
    title: string;
    data: LocationType[];
  };
};

export const transformDate = (date: Date) =>
  `${date.toLocaleDateString()}, ${date.toLocaleTimeString()}`;
export default function Item({item, deleteItem, section}: Props) {
  const styles = React.useContext(ThemeContext);
  const placeName = item?.location;
  const date = new Date(item?.time as number);
  const timeStr = transformDate(date);
  return (
    <View
      style={[
        {
          borderRadius: 10,
          backgroundColor: 'lightgray',
          flexDirection: 'row',
          alignItems: 'center',
        },
        styles.mar,
        styles.pad,
      ]}>
      <View style={{width: '80%'}}>
        <Text style={styles.p1}>{placeName}</Text>
        <Text style={[{marginTop: 5}, styles.p2]}>{timeStr}</Text>
      </View>
      {section.title !== 'Current Location' && (
        <View style={{width: '20%'}}>
          <Button title="remove" onPress={() => deleteItem(item?.id)} />
        </View>
      )}
    </View>
  );
}
