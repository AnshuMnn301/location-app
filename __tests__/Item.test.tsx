/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Item from '../src/components/Item';
import { fireEvent, render, screen } from '@testing-library/react-native';

it('renders correctly', () => {
    const mockDeleteItem =  jest.fn()
  const tree = renderer.create(<Item deleteItem={mockDeleteItem} section={{"data": [{"id": 3, "latlng": {"latitude": 17.6234567, "longitude": 78.77434}, "location": "South Hillview Drive, Milpitas, Santa Clara County, California, 95035, United States", "time": 1682594913918}, {"id": 2, "latlng": {"latitude": 17.6234567, "longitude": 78.77434}, "location": "South Hillview Drive, Milpitas, Santa Clara County, California, 95035, United States", "time": 1682594867802}, {"id": 1, "latlng": {"latitude": 17.6234567, "longitude": 78.77434}, "location": "Turkapalle, Turkapalle (M) mandal, Yadadri Bhuvanagiri, Telangana, India", "time": 1682594793690}], "title": "Previous Location"}} item={{"id": 1, "latlng": {"latitude": 17.6234567, "longitude": 78.77434}, "location": "Turkapalle, Turkapalle (M) mandal, Yadadri Bhuvanagiri, Telangana, India", "time": 1682594793690}}/>).toJSON();
  expect(tree).toMatchSnapshot()
});

describe('Test Item component', () => {
    beforeEach(() => {
        const mockDeleteItem =  jest.fn()
        render(<Item deleteItem={mockDeleteItem} section={{"data": [{"id": 3, "latlng": {"latitude": 17.6234567, "longitude": 78.77434}, "location": "South Hillview Drive, Milpitas, Santa Clara County, California, 95035, United States", "time": 1682594913918}, {"id": 2, "latlng": {"latitude": 17.6234567, "longitude": 78.77434}, "location": "South Hillview Drive, Milpitas, Santa Clara County, California, 95035, United States", "time": 1682594867802}, {"id": 1, "latlng": {"latitude": 17.6234567, "longitude": 78.77434}, "location": "Turkapalle, Turkapalle (M) mandal, Yadadri Bhuvanagiri, Telangana, India", "time": 1682594793690}], "title": "Previous Location"}} item={{"id": 1, "latlng": {"latitude": 17.6234567, "longitude": 78.77434}, "location": "Turkapalle, Turkapalle (M) mandal, Yadadri Bhuvanagiri, Telangana, India", "time": 1682594793690}}/>)
    })
    it('Delete selected location item', () => {
        fireEvent(screen.getByText('remove'), 'onPress', [])
    })
})