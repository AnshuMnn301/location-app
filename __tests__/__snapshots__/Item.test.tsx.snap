// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "borderRadius": 10,
        "flexDirection": "row",
      },
      Object {
        "margin": "2.5%",
      },
      Object {
        "padding": "2.5%",
      },
    ]
  }
>
  <View
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Turkapalle, Turkapalle (M) mandal, Yadadri Bhuvanagiri, Telangana, India
    </Text>
    <Text
      style={
        Array [
          Object {
            "marginTop": 5,
          },
          Object {
            "color": "black",
            "fontSize": 14,
          },
        ]
      }
    >
      27/4/2023, 4:56:33 pm
    </Text>
  </View>
  <View
    style={
      Object {
        "width": "20%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          remove
        </Text>
      </View>
    </View>
  </View>
</View>
`;
